cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/TreasuryDonation.hs

```diff
@@ -0,0 +1,149 @@
+        }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  PoolNode{poolRuntime} <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket poolRuntime
+  execConfig <- mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+  let socketPath = nodeSocketPath poolRuntime
+
+  epochStateView <- getEpochStateView configurationFile socketPath
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> unFile socketPath
+  H.note_ $ "Foldblocks config file: " <> unFile configurationFile
+
+  L.Coin currentTreasury <- getTreasuryValue epochStateView
+  H.note_ $ "currentTreasury: " <> show currentTreasury
+  currentTreasury H.=== 0 -- Treasury should initially be 0
+
+  let doOneDonation = doTreasuryDonation sbe execConfig work epochStateView wallet0
+
+  doOneDonation 0 Nothing 500
+  doOneDonation 1 Nothing 500_013
+  doOneDonation 2 Nothing (-497) -- Test donation that should fail because donation is negative
+  doOneDonation 3 (Just 1_234) (-497) -- Test donation that should fail because current treasury value is wrong
+
+doTreasuryDonation :: ()
+  => (HasCallStack, MonadCatch m, MonadTest m, MonadIO m, H.MonadAssertion m)
+  => ShelleyBasedEra era
+  -> H.ExecConfig
+  -> FilePath -- ^ Where temporary files can be stored
+  -> EpochStateView
+  -> PaymentKeyInfo-- ^ The key paying the fee
+  -> Int -- ^ The number of the call, used to create unique temporary file names. Starts at 0.
+  -> Maybe Int -- ^ The current treasury value to use. If unspecified, it will obtained from the node.
+  -> Int -- ^ The amount to donate
+  -> m ()
+doTreasuryDonation sbe execConfig work epochStateView wallet0 idx currentTreasury' treasuryDonation = do
+  currentTreasury <- 
+    case currentTreasury' of
+      Nothing -> do
+        v <- unCoin <$> getTreasuryValue epochStateView
+        H.note_ $ "currentTreasury: " <> show v
+        return v
+      Just x -> pure $ toInteger x
+
+  txBodyFp <- H.note $ work </> "treasury-donation-" <> show idx <> ".body"
+  signedTxFp <- H.note $ work </> "treasury-donation-" <> show idx <> ".signed"
+
+  txIn0 <- findLargestUtxoForPaymentKey epochStateView sbe wallet0
+
+  (exitCode, stdout, stderr) <- execCliAny execConfig
+    [ "conway", "transaction", "build"
+    , "--tx-in", Text.unpack $ renderTxIn txIn0
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr wallet0
+    , "--treasury-donation", show treasuryDonation
+    , "--out-file", txBodyFp
+    ]
+  when (not $ null stdout) (H.note_ stdout)
+  when (not $ null stderr) (H.note_ stderr)
+
+  (exitCode == ExitSuccess) H.=== (currentTreasury >= 0 && treasuryDonation >= 0)
+
+  case exitCode of
+    ExitFailure _ -> do
+      return ()
+    ExitSuccess -> do
+      H.noteM_ $ execCli' execConfig
+        [ "conway", "transaction", "view" , "--tx-file", txBodyFp ]
```

Comment:
This appears to be producing large output. Maybe it would be better to dump this into a file in the test workspace instead? Failed workspaces are preserved in GHA run.

Reply:
:+1: Changed like this:

```diff
@@ -104,6 +104,7 @@ doTreasuryDonation sbe execConfig work epochStateView wallet0 idx currentTreasur
       Just x -> pure $ toInteger x
 
   txBodyFp <- H.note $ work </> "treasury-donation-" <> show idx <> ".body"
+  txViewFp <- H.note $ work </> "treasury-donation-" <> show idx <> ".body.view.json"
   signedTxFp <- H.note $ work </> "treasury-donation-" <> show idx <> ".signed"
 
   txIn0 <- findLargestUtxoForPaymentKey epochStateView sbe wallet0
@@ -132,7 +133,8 @@ doTreasuryDonation sbe execConfig work epochStateView wallet0 idx currentTreasur
         -- greater or equal to zero.
         ct >= 0 && td >= 0
       H.noteM_ $ execCli' execConfig
-        [ "conway", "transaction", "view" , "--tx-file", txBodyFp ]
+        [ "conway", "transaction", "view" , "--tx-file", txBodyFp
+        , "--output-json", "--out-file", txViewFp]
```